nin/frontend/app/scripts/controllers/main.js

```diff
@@ -206,17 +206,17 @@ class MainCtrl {
           let graph = JSON.parse(event.content);
 
           for (let nodeInfo of graph) {
-            let node = demo.nm.createNode(nodeInfo);
-
             try {
+              let node = demo.nm.createNode(nodeInfo);
               demo.nm.insertOrReplaceNode(node);
             } catch (e) {
               // This hack only works due to not-yet received
               // nodes created through generate not having
               // any connections / friends.
               $timeout(function () {
+                let node = demo.nm.createNode(nodeInfo);
                 demo.nm.insertOrReplaceNode(node);
-              }, 100);
+              }, 200);
```

Comment:
Just re-moving the createnode line inside the try-catch didn't seem to help, but i can't remember if increasing the number to 200 helped either. Might need more testing.